packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -116,6 +123,18 @@ export const useCheckoutPage = props => {
         }
     }, [checkoutStep, setCheckoutStep]);
 
+    const revertPaymentInformationDone = useCallback(() => {
+        /**
+         * This step will reset the checkout step to PAYMENT.
+         *
+         * This function will be called when the placeOrder mutation
+         * has failed due to payment information issues. By resetting
+         * the step to PAYMENT, we will make the user re-enter the
+         * payment information.
+         */
+        setCheckoutStep(CHECKOUT_STEP.PAYMENT);
```

Comment:
Instead of threading a callback to the child you can just do this in a try/catch inside `handlePlaceOrder`.

Reply:
I am not sure how will throwing an error in the parent will propagate to the child.

Reply:
`placeOrderError` is still that same error that is thrown but my point is that logic specific to checkout page should be done within the checkout page callback and logic specific to error handling for payment information such as removing the nonce should be done down the line. You've got a combination of these two things, I'm just asking that you split them.

Reply:
Commented below in another thread about my thoughts.

Reply:
#2394 had logic in the place order catch block for setting review order button to false and setting checkout step to payment. If we're going to delegate the errors to the children for handling then we don't need the catch logic anymore (though we still should log the error for development purposes). So I would remove lines 139 and 140 in this file and perhaps replace them with a comment about how the child components handle submission errors.

Additionally, we also started passing `setCheckoutStep` and `resetShouldSubmit` to [payment information](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js#L141-L142).

Instead of writing the error handler here in checkout, could you define it in the `usePaymentInformation` talon? We don't need to define it here and pass it down as nothing else uses it.